sklearn/gaussian_process/gpr.py

```diff
@@ -236,6 +236,12 @@ def obj_func(theta, eval_gradient=True):
             # likelihood
             lml_values = list(map(itemgetter(1), optima))
             self.kernel_.theta = optima[np.argmin(lml_values)][0]
+            if np.isclose(self.kernel_.bounds,
+                          np.atleast_2d(self.kernel_.theta).T).any():
+                warnings.warn("Some parameters of the chosen kernel "
+                              "have been optimized to one of their bounds. "
+                              "Please broaden the bounds and re-do the fit",
+                              ConvergenceWarning)
```

Comment:
Is this warning enough ? I could try to tell exactly which parameters are poorly optimized but there are many different cases (e.g. mutiple kernels, multiple bands for one kernel, ...)
But I should be able to figure it out

Reply:
I don't think there's any harm doing the work to make this as informative as possible. What do you think of "The optimal value found for parameter {}={0.000} is close to the specified {upper/lower} bound {0.000}. {Increasing/Decreasing} the bound and calling fit again may find a better value." being repeated for each parameter..?